hector_mapping/src/HectorMappingRos.cpp

```diff
@@ -372,6 +381,32 @@ bool HectorMappingRos::mapCallback(nav_msgs::GetMap::Request  &req,
   return true;
 }
 
+bool HectorMappingRos::resetMapCallback(std_srvs::Trigger::Request  &req,
+                                        std_srvs::Trigger::Response &res)
+{
+  ROS_INFO("HectorSM Reset map service called");
+  slamProcessor->reset();
+  return true;
+}
+
+bool HectorMappingRos::pauseMapCallback(std_srvs::SetBool::Request  &req,
+                                        std_srvs::SetBool::Response &res)
+{
+  if (req.data && !pause_scan_processing_)
+  {
+    res.message = "Mapping paused";
+    ROS_INFO("HectorSM %s", res.message.c_str());
+  }
+  else if (!req.data && pause_scan_processing_)
+  {
+    res.message = "Mapping no longer paused";
+    ROS_INFO("HectorSM %s", res.message.c_str());
+  }
+  pause_scan_processing_ = req.data;
+  res.success = true;
+  return true;
```

Comment:
This will add a message if the pause status changed but no message if it stays the same?
Is this intended and what is the reasoning?
Just out of curiosity, I don't think it's necessary to return any message but if you do, why only on change?

Reply:
Well, the only reason that I am populating the "message" entry is because it comes with the "SetBool" service type, but it is quite useless from my experience...
It's up to you: I can make it always return a message, or never return a message, or only return when the state actually change (as it is implemented right now).
Just let me know your favorite solution, and I'll make it happen =)

Reply:
I think I would prefer no message since it's just a waste of bandwidth (not significant but still no point in transmitting useless data).

Reply:
you got it!